.circleci/config.yml

```diff
@@ -0,0 +1,88 @@
+---
+version: 2.1
+
+docker_auth: &docker_auth
+  username: $DOCKERHUB_USERNAME
+  password: $DOCKERHUB_PASSWORD
+
+defaults: &defaults
+  docker:
+    - image: greenpeaceinternational/circleci-base:latest
+      auth:
+        <<: *docker_auth
+  working_directory: /home/circleci/
+jobs:
+  clone-tokens-repo:
+    <<: *defaults
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run:
+          name: Clone planet4-design-tokens
+          command: |
+            mkdir tmp && cd "$_"
+            git clone -b PLANET-7002_implement-ci-automation https://github.com/greenpeace/planet4-design-tokens
+      - persist_to_workspace:
+          root: ./tmp/planet4-design-tokens
+          paths:
+            - ""
+  build-tokens:
+    <<: *defaults
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - attach_workspace:
+          at: ./tmp/planet4-design-tokens
+      - run:
+          name: Convert json tokens into a sass file
+          command: |
+            cd ./tmp/planet4-design-tokens
+            python3 ./bin/build.py
+      - persist_to_workspace:
+          root: ./tmp/planet4-design-tokens
+          paths:
+            - ""
+  release-tokens:
+    <<: *defaults
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - attach_workspace:
+          at: ./tmp/planet4-design-tokens
+      - run:
+          name: Create new release of tokens
+          command: |
+            python3 ./tmp/planet4-design-tokens/bin/release.py
+            mv ./tmp/planet4-design-tokens/src/_tokens.scss /home/circleci/tmp
+            ls -la /home/circleci/tmp
+      - persist_to_workspace:
+          root: ./tmp
+          paths:
+            - ""
+  promote-tokens-to-theme:
+    <<: *defaults
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - attach_workspace:
+          at: ./tmp
+      - run:
+          name: Promote tokens to master-theme
+          command: |
+            python3 ./tmp/planet4-design-tokens/bin/promote.py "./tmp"
+workflows:
+  main:
+    jobs:
+      - clone-tokens-repo
+      - build-tokens:
+          requires:
+            - clone-tokens-repo
+      - release-tokens:
```

Comment:
Shouldn't that be a manual step done by the Design Team? 🤔 So we don't have a PR being created for every push from Figma, but instead somene from the design team can come in this repo and create a new release. We could discuss that with the Design team.

But if we do want to create a master-theme PR for every push in the `main` branch of this repo, then we can simplify the workflow by not tagging at all. Just always building `_tokens.scss` from the latest commit on `main`.

Reply:
**About the paragraph #1** 
If the design team create a new release, how can we run the promote to the master theme job? Is it possible to first run the `build-tokens` job and after a release run `promote-to-master-theme`?

**About the paragraph #2**
I agree, but, don't we need to track versions to show differences between newest and latest ones? Or comparing between commits it's fine?

Thinking..
What if we create a new PR after the designers push a ne

Reply:
So here are the two options I see:

**First use case: Design team manually creating releases**

The whole CI workflow is only triggered on release/tags. This is easy to do with [filters](https://github.com/greenpeace/planet4-circleci/blob/main/.circleci/config.yml#L91-L95). No pipeline need to run when a new commit is pushed to `main` or any other branch. (Maybe we can add a linting job later).

1. A new release is created here.
2. A pipeline runs that has two jobs:
   - Build `_tokens.scss` out of `tokens.json`
   - Checkout master-theme and create a new PR

**Build on each `main` commit**

Same workflow as above but using a filter to run it only on `main` branch commits. The master-theme PR will have the new `_tokens.scss`, so yes that diff should be sufficient.